eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -196,19 +197,35 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         if (!$this->searchIndexer instanceof IncrementalIndexer) {
             $output->writeln(<<<EOT
-DEPRECATED:
-Running indexing against an Indexer that has not been updated to use IncrementalIndexer abstract.
-
-Options that won't be taken into account:
-- since
-- content-ids
-- subtree
-- processes
-- no-purge
-EOT
-            );
+                DEPRECATED:
+                Running indexing against an Indexer that has not been updated to use IncrementalIndexer abstract.
+                
+                Options that won't be taken into account:
+                - since
+                - content-ids
+                - subtree
+                - processes
+                - no-purge
+            EOT);
             $this->searchIndexer->createSearchIndex($output, (int) $iterationCount, !$commit);
         } else {
+            $io = new SymfonyStyle($input, $output);
+            $xdebugState = \extension_loaded('xdebug') ? 'enabled' : 'disabled';
+            $memoryLimit = ini_get('memory_limit');
+
+            $io->warning(<<<EOT
+                For optimal performance, before running this command, make sure that:
+                - the xdebug extension is disabled (you have it $xdebugState),
+                - you're running the command in "prod" environment (default: dev), 
+                - memory limit for big databases is set to "-1" or an adequately high value (your value: $memoryLimit),
+                - --iteration-count is low enough (default: 50),
+                - number of processes for parallel batch operations is high enough (default: 'auto' is a good choice).
+            EOT);
+
+            if (!$io->confirm('Continue?', true)) {
+                return 0;
+            }
```

Comment:
Maybe move to helper function, but this is something what I had in mind here.

Could even consider only showing it `if (\in_array($input->getOption('processes'), ['0', '1']) && \extension_loaded('xdebug') )`

```suggestion

            if (\in_array($input->getOption('processes'), ['0', '1'])) {
                $io = new SymfonyStyle($input, $output);
                $xdebugState = \extension_loaded('xdebug') ? '<error>enabled</error>' : '<info>disabled</info>';
                $memoryLimit = ini_get('memory_limit');

                $io->warning(<<<EOT
                    It's not recommended to run this command in single process mode with a large dataset!

                    To avoid running into memory issues, make sure that:
                    - the xdebug extension is disabled (you have it $xdebugState),
                    - you're running the command in "prod" environment (default: dev), 
                    - memory limit for big databases is set to "-1" or an adequately high value (your value: $memoryLimit),
                    - <info>--iteration-count</info> is low enough (default: 50),
                    - number of processes for parallel batch operations is high enough (default: 'auto' is a good choice).
                EOT);

                if (!$io->confirm('Continue?', true)) {
                    return 0;
                }
            }
```

Reply:
Changed in https://github.com/ezsystems/ezplatform-kernel/pull/138/commits/ee7164bd7d3af245ddbd52a6d4b14a2982921641.